algo/src/main/java/org/neo4j/graphalgo/PageRankProc.java

```diff
@@ -86,11 +87,18 @@
             @Name(value = "label", defaultValue = "") String label,
             @Name(value = "relationship", defaultValue = "") String relationship,
             @Name(value = "config", defaultValue = "{}") Map<String, Object> config) {
-        config.put(CONFIG_WRITE, Boolean.FALSE);
+        if ((boolean) config.getOrDefault(CONFIG_WRITE, DEFAULT_WRITE)) {
+            Map<String, Object> newConfig = new HashMap<>(config);
+            newConfig.put(CONFIG_WRITE, Boolean.FALSE);
+            config = newConfig;
+        }
         return pageRank(label, relationship, config);
```

Comment:
for later, the internal methods that are called should just have spelled out parameters, so we just pass the right ones along.